source/isaaclab/isaaclab/sim/simulation_context.py

```diff
@@ -256,6 +256,9 @@ def __init__(self, cfg: SimulationCfg | None = None):
         #   you can reproduce the issue by commenting out this line and running the test `test_articulation.py`.
         self._gravity_tensor = torch.tensor(self.cfg.gravity, dtype=torch.float32, device=self.cfg.device)
 
+        # define a global variable to store the exceptions raised in the callback stack
+        builtins.ISAACLAB_CALLBACK_EXCEPTION = None
```

Comment:
This should be a list incase multiple callbacks failed? For instance, when the contact sensor callback fails because asset creation failed, the current code will overwrite the exception from asset class. The user will only get the error of contact sensor failure which doesn't help them get to the true issue.

Reply:
Hm print all the errors through omni.log.error and raise the first one? But yeah you are right. Didn't see there was a "if none" check happening.